easybuild/easyblocks/q/quantumespresso.py

```diff
@@ -61,132 +63,269 @@ def __init__(self, *args, **kwargs):
+            self.extra_libs.append(" -lxcf90 -lxc")
 
+        self._dflags += ["-D__LIBXC"]
+
+    def _add_hdf5(self):
+        """Add HDF5 support to the build."""
         hdf5 = get_software_root("HDF5")
-        if hdf5:
-            self.cfg.update('configopts', '--with-hdf5=%s' % hdf5)
-            dflags.append(" -D__HDF5")
-            hdf5_lib_repl = '-L%s/lib -lhdf5hl_fortran -lhdf5_hl -lhdf5_fortran -lhdf5 -lsz -lz -ldl -lm' % hdf5
-            repls.append(('HDF5_LIB', hdf5_lib_repl, False))
+        if not hdf5:
+            return
+        self.cfg.update('configopts', f'--with-hdf5={hdf5}')
+        self._dflags += ["-D__HDF5"]
+        hdf5_lib_repl = f'-L{hdf5}/lib -lhdf5hl_fortran -lhdf5_hl -lhdf5_fortran -lhdf5 -lsz -lz -ldl -lm'
+        self._repls += [('HDF5_LIB', hdf5_lib_repl, False)]
+
+        if LooseVersion(self.version) >= LooseVersion("6.2.1"):
+            pass
+        else:
+            # Should be experimental in 6.0 but gives segfaults when used
+            raise EasyBuildError("HDF5 support is only available in QuantumESPRESSO 6.2.1 and later")
 
+    def _add_elpa(self):
+        """Add ELPA support to the build."""
         elpa = get_software_root("ELPA")
-        if elpa:
-            if not self.cfg['with_scalapack']:
-                raise EasyBuildError("ELPA requires ScaLAPACK but 'with_scalapack' is set to False")
-
-            elpa_v = get_software_version("ELPA")
-            if LooseVersion(self.version) >= LooseVersion("7"):
-                # NOTE: from version 7, there are only three __ELPA flags,
-                # - __ELPA for ELPA releases 2018.11 and beyond;
-                # - __ELPA_2016 for ELPA releases 2016.11, 2017.x and 2018.05;
-                # - __ELPA_2015 for ELPA releases 2015.x and 2016.05;
-                # see https://github.com/QEF/q-e/commit/351f4871fee3c8045d75592dde606b2279b08e02
-                if LooseVersion(elpa_v) >= LooseVersion("2018.11"):
-                    dflags.append('-D__ELPA')
-                elif LooseVersion(elpa_v) >= LooseVersion("2016.11"):
-                    dflags.append('-D__ELPA_2016')
-                else:
-                    dflags.append('-D__ELPA_2015')
-
-                elpa_min_ver = "2015"
-            elif LooseVersion(self.version) >= LooseVersion("6"):
-                # NOTE: Quantum Espresso 6.x should use -D__ELPA_<year> for corresponding ELPA version
-                # However for ELPA VERSION >= 2017.11 Quantum Espresso needs to use ELPA_2018
-                # because of outdated bindings. See: https://xconfigure.readthedocs.io/en/latest/elpa/
-                if LooseVersion("2018") > LooseVersion(elpa_v) >= LooseVersion("2017.11"):
-                    dflags.append('-D__ELPA_2018')
-                else:
-                    # get year from LooseVersion
-                    elpa_year_v = elpa_v.split('.')[0]
-                    dflags.append('-D__ELPA_%s' % elpa_year_v)
+        if not elpa:
+            return
```

Comment:
We generally don't use inline return statements as they can be overlooked when reading, can be avoided here pretty easily

Reply:
Fixed in 96ededc